XCUITests/FxScreenGraph.swift

```diff
@@ -242,77 +284,94 @@ func createScreenGraph(_ app: XCUIApplication, url: String = "https://www.mozill
+        scene.gesture(to: Intro_FxASignin) {
+            let introScrollView = app.scrollViews["IntroViewController.scrollView"]
+            for _ in 1...4 {
+                introScrollView.swipeLeft();
+            }
+            app.buttons["Sign in to Firefox"].tap()
+        }
+        scene.backAction = {
+            introScrollView.swipeLeft()
+            startBrowsingButton.tap()
+        }
+        
+    }
+    
+    map.createScene(Intro_FxASignin) { scene in
+        // Note: if a browser tab was open beforehand, this will go back to
+        // BrowserTab. In such case, manual redirection with nowAt() is needed
+        scene.tap(app.navigationBars["Client.FxAContentView"].buttons["Done"], to: HomePanelsScreen)
+    }
 
     map.createScene(PrivateTabTray) { scene in
-        scene.tap(app.buttons["TabTrayController.menuButton"], to: TabTrayMenu)
-        scene.tap(app.buttons["TabTrayController.addTabButton"], to: NewPrivateTabScreen)
+        scene.tap(app.buttons["TabTrayController.addTabButton"], to: PrivateHomePanelsScreen)
         scene.tap(app.buttons["TabTrayController.maskButton"], to: TabTray)
     }
 
     map.createScene(TabTray) { scene in
-        scene.tap(app.buttons["TabTrayController.menuButton"], to: TabTrayMenu)
-        scene.tap(app.buttons["TabTrayController.addTabButton"], to: NewTabScreen)
+        scene.tap(app.buttons["TabTrayController.addTabButton"], to: HomePanelsScreen)
         scene.tap(app.buttons["TabTrayController.maskButton"], to: PrivateTabTray)
     }
 
-    map.createScene(TabTrayMenu) { scene in
-        scene.gesture(to: SettingsScreen) {
-            // XXX The element is fails the existence test, so we tap it through the gesture() escape hatch.
-            app.tables.cells["Settings"].tap()
-        }
-
-        scene.backAction = closeMenuAction
-        scene.dismissOnUse = true
-    }
-
     map.createScene(BrowserTab) { scene in
         scene.tap(app.textFields["url"], to: URLBarOpen)
         scene.tap(app.buttons["TabToolbar.menuButton"], to: BrowserTabMenu)
         if map.isiPad() {
             scene.tap(app.buttons["TopTabsViewController.tabsButton"], to: TabTray)
         } else {
-            scene.tap(app.buttons["URLBarView.tabsButton"], to: TabTray)
+            scene.gesture(to: TabTray) {
+                if (app.buttons["TabToolbar.tabsButton"].exists) {
+                    app.buttons["TabToolbar.tabsButton"].tap()
+                } else {
+                    app.buttons["URLBarView.tabsButton"].tap()
+                }
+            }
         }
+        scene.tap(app.buttons["TabLocationView.pageOptionsButton"], to: PageOptionsMenu)
+        
+        scene.backAction = backBtnBackAction
     }
-
+    
+    // make sure after the menu action, navigator.nowAt() is used to set the current state
+    map.createScene(PageOptionsMenu) {scene in
+        scene.tap(app.tables["Context Menu"].staticTexts["Find in Page"], to: FindInPage)
```

Comment:
Please add `accessibilityIdentifier` to any views that need them.

Reply:
For the context menu, I think I can use self.iconString of PhotonActionSheetItem.  I'm a bit wary of creating an acc. id. element and populating them artificially